src/main/java/core/basesyntax/sevice/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.sevice.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.Strategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final Strategy strategy;
+
+    public ShopServiceImpl(Strategy strategy) {
+        this.strategy = strategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> fruitTransactions) {
+        for (FruitTransaction fruitTransaction: fruitTransactions) {
+            strategy.get(fruitTransaction.getOperation())
+                    .processOperation(fruitTransaction);
+        }
+    }

```

Comment:
The `ShopServiceImpl` class is tightly coupled with the `Strategy` interface, which could be a violation of the SOLID principles. Consider making the service independent and using dependency injection to pass in a map of strategies to handle different operations, which can be called in the `main()` method instead.